src/client/mod.rs

```diff
@@ -31,10 +31,14 @@ impl<T> Grpc<T> {
         Grpc { inner }
     }
 
-    pub fn poll_ready<R>(&mut self) -> Poll<(), ::Error<T::Error>>
-    where T: HttpService<R> {
+    pub fn poll_ready<R>(&mut self) -> Poll<(), ::Status>
+    where
+        T: HttpService<R>,
+    {
         self.inner.poll_ready()
-            .map_err(::Error::Inner)
+            .map_err(|_| {
+                ::Status::unknown("inner client service poll_ready error".into())
```

Comment:
I think this is the thing that made me think doing this is difficult; here you are throwing away error information, no?

Reply:
That's true. I tried requiring `HttpService<Error = impl Into<Status>>`, but it was irritating with the examples and interop.

I'm going to try just requiring `Error: std::error::Error`, since that can be used to downcast to `Status`, or otherwise make an `UNKNOWN` status with the error as the message.